### PR TITLE
Updates for evening of Feb 20

### PR DESCRIPTION
**Pull request by PaulFenton**

### CODE REVIEW

File: A1-2015/Library/FrameFilterFramework.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+public class FrameFilterFramework extends FilterFramework {
+
+    // add functions to read & write frames
+    Frame readFrame(PipedInputStream input) throws EndOfStreamException {
```
Comment (darioalessandro) [inline A1-2015/Library/FrameFilterFramework.java:9]: :+1:

File: A1-2015/Library/FramePrinterSink.java [hunk 2]
```diff
@@ -21,7 +21,7 @@ public void run() {
 			outStream.writeUTF(this.header);
 			while (true) {
 				try {
-					Frame m = ReadFrame();
+					Frame m = readFrame(this.InputReadPort);
```
Comment (darioalessandro) [inline A1-2015/Library/FramePrinterSink.java:24]: :+1:

File: A1-2015/Library/FrameSmoothFilter.java [hunk 3]
```diff
@@ -43,21 +43,25 @@ public void run() {
...
-				try {
-					ObjectOutputStream output = new ObjectOutputStream(this.OutputWritePort);
-					output.writeObject(transformedCurrent);
-					previous = transformedCurrent;
-					current = next;
-				}
-				catch (IOException e) {
-					e.printStackTrace();
-				}
+				//Frame transformedCurrent = frameSmoothFilterCallback.smoothCurrentFrame(next, current, previous);
```
Comment (darioalessandro) [inline A1-2015/Library/FrameSmoothFilter.java]: I think this line can go.

File: A1-2015/Library/SplitFilter.java [hunk 4]
```diff
@@ -1,66 +1,66 @@
...
+                        e.printStackTrace();
+                    }
+                }
+                ClosePorts();
+                break;
+            }
+        }
+    }
+
+    /*void writeFrame(Frame frame, PipedOutputStream output) {
```
Comment (darioalessandro) [inline A1-2015/Library/SplitFilter.java]: I think this can go if it is not needed

File: A1-2015/Library/TransformFrameCallback.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
...
-
-public interface TransformFrameCallback {
-    public Optional<Frame> transform(Frame frame, List<Frame> lastNSamples);
-}
+import java.util.Optional;
+import java.util.List;
+
+public interface TransformFrameCallback {
+    public Optional<Frame> transform(Frame frame, List<Frame> lastNSamples);
+    //public Frame transform(Frame frame, List<Frame> lastNSamples);
```
Comment (darioalessandro) [inline A1-2015/Library/TransformFrameCallback.java]: this function can be removed

File: A1-2015/Library/TransformFrameFilter.java [hunk 6]
```diff
@@ -34,7 +34,7 @@ public void run() {
 		System.out.print( "\n" + this.getName() + "::TransformFrameFilter ");
 		while (true) {
 			try {
-				transformFrameCallback.transform(ReadFrame(), Collections.unmodifiableList(lastNFrames))
+				transformFrameCallback.transform(readFrame(this.InputReadPort), Collections.unmodifiableList(lastNFrames))
```
Comment (darioalessandro) [inline A1-2015/Library/TransformFrameFilter.java:37]: :+1:

File: A1-2015/SystemBOOO/SystemB.java [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 /**
  * Pipe-and-filter network that does what System A does but includes pressure data in the
- * output. In addition, System B should filter “wild points” out of the data stream for pressure
+ * output. In addition, System B should filter “wild points�? out of the data stream for pressure
```
Comment (darioalessandro) [inline A1-2015/SystemBOOO/SystemB.java]: There's some encoding issue here, the quotes look like ``` â€œwild pointsâ€?```

**Review by darioalessandro — COMMENTED** [review]
I did not read it completely yet but I have some comments.

File: A1-2015/SystemBOOO/SystemB.java (hunk 7 above)
Comment (darioalessandro) [inline A1-2015/SystemBOOO/SystemB.java]: same thing with quotes

File: A1-2015/SystemC/MergeFilter.java [hunk 8]
```diff
@@ -32,40 +34,48 @@ public void run() {
...
-                    WriteFrameOutput(this.OutputWritePort, newestInputFrame);
-                }
-                
-            }
-          }else {
-              if(this.InputFilter.isAlive()){
-                  while(this.InputFilter.isAlive()){
-                      newestInputFrame = ReadFrameInput(this.InputReadPort);
-                      WriteFrameOutput(this.OutputWritePort, newestInputFrame);
+          if(inputStreamActive){
```
Comment (darioalessandro) [inline A1-2015/SystemC/MergeFilter.java:35]: does this boolean consider all the inputStreams and filters?

File: A1-2015/SystemC/MergeFilter.java [hunk 9]
```diff
@@ -1,5 +1,7 @@
 import java.io.PipedInputStream;
 
+import sun.font.TrueTypeFont;
```
Comment (darioalessandro) [inline A1-2015/SystemC/MergeFilter.java]: Thinks looks like some internal library, what do we need it for?

File: A1-2015/SystemC/SystemA.java [hunk 10]
```diff
@@ -0,0 +1,162 @@
+import java.text.DecimalFormat;
+import java.text.SimpleDateFormat;
+import java.util.Optional;
+
+import sun.invoke.empty.Empty;
+
+public class SystemA {
```
Comment (darioalessandro) [inline A1-2015/SystemC/SystemA.java]: how comes SystemA is in system C? should this be named SystemC?

File: A1-2015/SystemC/SystemA.java [hunk 11]
```diff
@@ -0,0 +1,162 @@
+import java.text.DecimalFormat;
+import java.text.SimpleDateFormat;
+import java.util.Optional;
+
+import sun.invoke.empty.Empty;
```
Comment (darioalessandro) [inline A1-2015/SystemC/SystemA.java]: I think this is a different empty than Optional.empty()

**Review by darioalessandro — COMMENTED** [review]
Part 2/2

File: A1-2015/SystemC/MergeFilter.java (hunk 8 above)
Comment (PaulFenton) [reply A1-2015/SystemC/MergeFilter.java:35]: I should probably rename it, it refers to which of the two streams the algorithm is currently reading from, it switches back and forth

File: A1-2015/SystemC/SystemA.java (hunk 10 above)
Comment (PaulFenton) [reply A1-2015/SystemC/SystemA.java]: Yes

File: A1-2015/SystemBOOO/SystemB.java [hunk 12]
```diff
@@ -82,16 +82,16 @@ public static void main(String argv[]) {
         // TODO: handle scenario where the first frame and last frames have a wild point.
         FrameSmoothFilter smoothFilter = new FrameSmoothFilter((next, current, previous) -> {
```
Comment (darioalessandro) [inline A1-2015/SystemBOOO/SystemB.java:83]: @PaulFenton what do you think?

**Review by darioalessandro — APPROVED** [review]
LGTM

**Review by PaulFenton — COMMENTED** [review]
looks good

**Pull request merged**